processor/metricstransformprocessor/datapoint_aggregation.go

```diff
@@ -0,0 +1,248 @@
+				Value:     intPoint,
+			})
+		case metricspb.MetricDescriptor_CUMULATIVE_DOUBLE, metricspb.MetricDescriptor_GAUGE_DOUBLE:
+			doublePoint := mtp.computeDouble(points, aggrType)
+			newPoints = append(newPoints, &metricspb.Point{
+				Timestamp: timestamp,
+				Value:     doublePoint,
+			})
+		case metricspb.MetricDescriptor_CUMULATIVE_DISTRIBUTION, metricspb.MetricDescriptor_GAUGE_DISTRIBUTION:
+			if aggrType != Sum {
+				mtp.logger.Warn("Distribution data can only be aggregated by taking the sum")
+				break
+			}
+			distPoints := mtp.computeDistribution(points)
+			for _, p := range distPoints {
+				newPoints = append(newPoints, &metricspb.Point{
+					Timestamp: timestamp,
+					Value:     p,
+				})
+			}
+
+		}
+	}
+	return newPoints
+}
+
+// computeInt64 attempts to merge int64 points into one point based on the provided aggrType
+// Returns the aggregated int64 value
+func (mtp *metricsTransformProcessor) computeInt64(points []*metricspb.Point, aggrType AggregationType) *metricspb.Point_Int64Value {
+	intVal := int64(0)
+	if len(points) > 0 {
+		intVal = points[0].GetInt64Value()
+	}
+	for _, p := range points[1:] {
+		if aggrType == Sum || aggrType == Mean {
+			intVal += p.GetInt64Value()
+		} else if aggrType == Max {
+			intVal = mtp.maxInt64(intVal, p.GetInt64Value())
+		} else if aggrType == Min {
+			intVal = mtp.minInt64(intVal, p.GetInt64Value())
+		}
+	}
+	if aggrType == Mean {
+		intVal /= int64(len(points))
+	}
+	return &metricspb.Point_Int64Value{Int64Value: intVal}
+}
+
+// computeDouble attempts to merge double points into one point based on the provided aggrType
+// Returns the aggregated double value
+func (mtp *metricsTransformProcessor) computeDouble(points []*metricspb.Point, aggrType AggregationType) *metricspb.Point_DoubleValue {
+	doubleVal := float64(0)
+	if len(points) > 0 {
+		doubleVal = points[0].GetDoubleValue()
+	}
+	for _, p := range points[1:] {
+		if aggrType == Sum || aggrType == Mean {
+			doubleVal += p.GetDoubleValue()
+		} else if aggrType == Max {
+			doubleVal = math.Max(doubleVal, p.GetDoubleValue())
+		} else if aggrType == Min {
+			doubleVal = math.Min(doubleVal, p.GetDoubleValue())
+		}
+	}
+	if aggrType == Mean {
+		doubleVal /= float64(len(points))
+	}
+	return &metricspb.Point_DoubleValue{DoubleValue: doubleVal}
+}
+
+// computeDistribution attempts to merge distribution points into one point
+// Returns the aggregated distribution value or values if there are mismatching bounds
+func (mtp *metricsTransformProcessor) computeDistribution(points []*metricspb.Point) []*metricspb.Point_DistributionValue {
+	distVals := make([]*metricspb.Point_DistributionValue, 0)
+	boundsToPoints := mtp.groupPointsByBounds(points)
+	for _, v := range boundsToPoints {
+		var distVal *metricspb.DistributionValue
+		for _, p := range v {
+			if distVal == nil {
+				distVal = p.GetDistributionValue()
+				continue
+			}
+			distVal = mtp.computeDistVals(distVal, p.GetDistributionValue())
```

Comment:
As above, name should be merge, or maybe aggregate:
```suggestion
			distVal = mtp.mergeDistVals(distVal, p.GetDistributionValue())
```